packages/core/package.json

```diff
@@ -56,7 +56,9 @@
     "rslib": "npm:@rslib/core@0.4.0",
     "rslog": "^1.2.3",
     "tsconfck": "3.1.4",
-    "typescript": "^5.7.3"
+    "typescript": "^5.7.3",
+    "typia": "^7.6.2",
```

Comment:
`typia` has so many dependencies, this is a bit hard to accept...
![image](https://github.com/user-attachments/assets/fcf76eb8-9fce-4723-9c27-8cbacb9325f2)

https://pkg-graph.info/typia

Reply:
It indeed has too many dependencies, but it is tree-shakable and we are using it as `devDependencies`.

The bundle size of Rslib is increased from **106.8 kB** to **2484.4 kB**. But the size change mainly produced by the validation functions generated by `typia`:

<img width="1245" alt="image" src="https://github.com/user-attachments/assets/3d4d41cc-8c3c-4aa3-8c31-a9f9b1de7c78" />

As you can see, the `src/validate.ts` has **2.33 MB** in stats. And we can make validation opt-in and lazy loaded.

For end users, no additional dependencies need to be installed, and the bundle size of `@rslib/core` increases by only approximately 2.5 MB.

Reply:
If we make the validation opt-in, how can users enable the validation?

Reply:
E.g.: if the `TypiaRspackPlugin`(which transforms `src/validation.ts`) is commented out, the bundle size of `@rslib/core` in this PR is only **108.4 kB**.

<img width="681" alt="image" src="https://github.com/user-attachments/assets/08036caf-b472-4ce0-8a08-a064e8166c4f" />

Reply:
> If we make the validation opt-in, how can users enable the validation?

Maybe through a environment variable?

```js
// rslib.config.js
process.env['RSLIB_CONFIG_VALIDATION'] = true

export default {}
```

Or maybe a `strictDefineConfig` API?

```js
import {strictDefineConfig} from '@rslib/core'

export default strictDefineConfig({})
```

But I would prefer enabling by default. And opt-out using environment variables.

Reply:
If it's opt-in, I don't think most users will actively enable it...

If it's opt-out, the performance overhead doesn't seem worth it, we can write code to do some validation manually.

Reply:
Sure, I'm closing this PR. Thank you for your review